src/Mado/QueryBundle/Tests/Queries/QueryBuilderFactoryTest.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Mado\QueryBundle\Tests\Queries;
+
+use Mado\QueryBundle\Queries\QueryBuilderFactory;
+use PHPUnit\Framework\TestCase;
+
+class QueryBuilderFactoryTest extends TestCase
+{
+    public function testGetAvailableFilters()
+    {
+        $entityManager = $this
+            ->getMockBuilder('Doctrine\ORM\EntityManager')
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $queryBuilderFactory = new QueryBuilderFactory($entityManager);
+
+        $expectedFilters = [
+            'eq' => [
+                'meta' => '=',
+            ],
+            'neq' => [
+                'meta' => '!=',
+            ],
+            'gt' => [
+                'meta' => '>',
+            ],
+            'gte' => [
+                'meta' => '>=',
+            ],
+            'lt' => [
+                'meta' => '<',
+            ],
+            'lte' => [
+                'meta' => '<=',
+            ],
+            'startswith' => [
+                'meta' => 'LIKE',
+                'substitution_pattern' => '{string}%'
+            ],
+            'contains' => [
+                'meta' => 'LIKE',
+                'substitution_pattern' => '%{string}%'
+            ],
+            'notcontains' => [
+                'meta' => 'NOT LIKE',
+                'substitution_pattern' => '%{string}%'
+            ],
+            'endswith' => [
+                'meta' => 'LIKE',
+                'substitution_pattern' => '%{string}'
+            ],
+            'list' => [
+                'meta' => 'IN',
+                'substitution_pattern' => '({string})',
+            ],
+            'field_eq' => [
+                'meta' => '=',
+            ],
+        ];
+
+        $availableFilters = $queryBuilderFactory->getValueAvailableFilters();
+
+        $this->assertEquals($expectedFilters, $availableFilters);
```

Comment:
If `Operators::getOperators()` will change this test will fail. To make this test more robust replace array definition with:

```php
$expectedFilters = Operators::getOperators();
```

With this fix, for any change of `Operators::getOperators()` this test will remain green.

Reply:
Mmmm you are not testing Operators: you are testing QueryBuilderFactory. 

    $this->assertEquals(
        Operators::getOperators(),
        $queryBuilderFactory->getValueAvailableFilters()
    );